src/main/java/com/scalar/db/storage/jdbc/query/UpdateQuery.java

```diff
@@ -0,0 +1,131 @@
+package com.scalar.db.storage.jdbc.query;
+
+import com.scalar.db.api.ConditionalExpression;
+import com.scalar.db.io.Key;
+import com.scalar.db.io.Value;
+import com.scalar.db.storage.jdbc.Table;
+
+import javax.annotation.Nullable;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.scalar.db.storage.jdbc.query.QueryUtils.getOperatorString;
+
+public class UpdateQuery extends AbstractQuery {
+
+  public static class Builder {
+    private final Table table;
+    private Key partitionKey;
+    @Nullable private Key clusteringKey;
+    private Map<String, Value> values;
+    @Nullable List<ConditionalExpression> otherConditions;
+
+    Builder(Table table) {
+      this.table = table;
+    }
+
+    public Builder set(Map<String, Value> values) {
+      this.values = values;
+      return this;
+    }
+
+    public Builder where(Key partitionKey, @Nullable Key clusteringKey) {
+      return where(partitionKey, clusteringKey, null);
+    }
+
+    public Builder where(
+        Key partitionKey,
+        @Nullable Key clusteringKey,
+        @Nullable List<ConditionalExpression> otherConditions) {
+      this.partitionKey = partitionKey;
+      this.clusteringKey = clusteringKey;
+      this.otherConditions = otherConditions;
+      return this;
+    }
+
+    public UpdateQuery build() {
+      if (partitionKey == null || values == null) {
+        throw new IllegalStateException("partitionKey or values is null.");
+      }
+      return new UpdateQuery(this);
+    }
+  }
+
+  private final Table table;
+  private final Key partitionKey;
+  @Nullable private final Key clusteringKey;
+  private final Map<String, Value> values;
+  @Nullable private final List<ConditionalExpression> otherConditions;
+
+  private UpdateQuery(Builder builder) {
+    table = builder.table;
+    partitionKey = builder.partitionKey;
+    clusteringKey = builder.clusteringKey;
+    values = builder.values;
+    otherConditions = builder.otherConditions;
+  }
+
+  @Override
+  protected String sql() {
+    return "UPDATE " + table + " SET " + makeSetSQLString() + " WHERE " + makeConditionSQLString();
```

Comment:
Guave's Strings.Joiner might be useful here to handle spaces nicely. (Also, it's a minor thing but, I'm sure it is more efficient than simple string concatenation, which is supposed to be not efficient)
```suggestion
    return Joiner.join("UPDATE", table, "SET", makeSetSQLString(), "WHERE", makeConditionSQLString());
```
https://github.com/google/guava/wiki/StringsExplained#joiner